deft.el

```diff
@@ -1705,25 +1705,28 @@ Otherwise, quick create a new file."
 (declare-function org-add-link-type "org")
 (declare-function org-open-file-with-emacs "org")
 
+(defun deft-id-at-point ()
+  (substring (expand-file-name (deft-filename-at-point))
+             (1+ (length (expand-file-name deft-directory))))
+  )
 (defun org-deft-store-link ()
   "Store the Deft widget at point as an org-mode link."
   (when (equal major-mode 'deft-mode)
-    (let ((link (concat "deft:" (substring (deft-filename-at-point)
-                                           (1+ (length deft-directory)))))
+    (let ((link (concat "deft:" (deft-id-at-point)))
           (title (deft-file-title (deft-filename-at-point))))
       (org-store-link-props
        :type "deft"
        :link link
        :description title))))
 
 (with-eval-after-load 'org
-  (org-add-link-type
-   "Deft"
-   (lambda (handle)
-     (org-open-file-with-emacs
-      (expand-file-name handle deft-directory)))))
+  (org-link-set-parameters
+   "deft" :follow 'follow-deft :store 'org-deft-store-link
+   ))
 
-(add-hook 'org-store-link-functions 'org-deft-store-link)
+(defun follow-deft (handle)
```

Comment:
Since `follow-deft` is in the global "namespace," maybe naming this something like `deft--org-follow-link`would be better.